src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.cs

```diff
@@ -51,6 +43,14 @@ internal sealed partial class RemoteWorkspace : Workspace
         /// </summary>
         private int _currentRemoteWorkspaceVersion = -1;
 
+        /// <summary>
+        /// Mapping from solution checksum to to the solution computed for it.  This is used so that we can hold a
+        /// solution around as long as the checksum for it is being used in service of some feature operation (e.g.
+        /// classification).  As long as we're holding onto it, concurrent feature requests for the same checksum can
+        /// share the computation of that particular solution and avoid duplicated concurrent work.
+        /// </summary>
+        private readonly Dictionary<Checksum, (int refCount, AsyncLazy<Solution> lazySolution)> _checksumToRefCountAndLazySolution = new();
```

Comment:
>_checksumToRefCountAndLazySolution [](http://example.com/codeflow?start=96&length=34)

perhaps better name: `_availableSolutions`

Reply:
Hrmm... i'm ok with the name as is.  _availableSolutions is a bit too vague for me.